client-js/main/client/parser/well-known-parsers.js

```diff
@@ -0,0 +1,243 @@
+
+  fromObject(object) {
+    let uInt64Value = new wrappers.UInt64Value();
+    uInt64Value.setValue(object);
+    return uInt64Value;
+  }
+}
+
+export class ListValueParser extends ObjectParser {
+
+  fromObject(object) {
+    let listValue = new struct.ListValue;
+    object.forEach(
+      function callback(currentValue, index, array) {
+        let valueParser = new ValueParser();
+        array[index] = valueParser.parse(currentValue);
+      }
+    );
+    listValue.setValuesList(object);
+    return listValue;
+  }
+}
+
+export class ValueParser extends ObjectParser {
+
+  fromObject(object) {
+    let result = new struct.Value();
+    if (object === null) {
+      result.setNullValue(struct.NullValue.NULL_VALUE);
+    } else if (typeof object === "number") {
+      result.setNumberValue(object);
+    } else if (typeof object === "string") {
+      result.setStringValue(object);
+    } else if (typeof object === "boolean") {
+      result.setBoolValue(object);
+    } else if (Array.isArray(object)) {
+      let parser = new ListValueParser(object);
+      let listValue = parser.parse(object);
+      result.setListValue(listValue);
+    } else {
+      // Is a Struct, unhandled for now.
+    }
+    return result;
+  }
+}
+
+export class EmptyParser extends ObjectParser {
+
+  fromObject(object) {
+    let emptyValue = new Empty();
+    return emptyValue;
+  }
+}
+
+export class TimestampParser extends ObjectParser {
+
+  fromObject(object) {
+    let date = new Date(object);
+    let result = new Timestamp();
+    result.fromDate(date);
+    return result;
+  }
+}
+
+export class DurationParser extends ObjectParser {
+
+  fromObject(object) {
+    object = object.substring(0, object.length - 1);
+    let values = object.split(".");
+    let result = new Duration();
+    if (values.length === 1) {
+      result.setSeconds(values[0]);
+    } else if (values.length === 2) {
+      result.setSeconds(values[0]);
+      let nanos = values[1];
+      for (let i = 0; i < 9 - nanos.length; i++) {
+        nanos += "0";
+      }
+      let nanosNumber = parseInt(nanos, 10);
+      result.setNanos(nanosNumber);
+    }
+    return result;
+  }
+}
+
+export class FieldMaskParser extends ObjectParser {
+
+  fromObject(object) {
+    let fieldMask = new FieldMask();
+    fieldMask.setPathsList(object.split(","));
+    return fieldMask;
+  }
+}
+
+export class AnyParser extends ObjectParser {
+
+  fromObject(object) {
+    const typeUrl = object["@type"];
+    const isWellKnown = wellKnownParsers.has(typeUrl);
+    const packedValue = isWellKnown
+      ? object["value"]
+      : object;
+    const parser = TypeParsers.parserFor(typeUrl);
+    const messageValue = parser.fromObject(packedValue);
+    const bytes = messageValue.serializeBinary();
+    const anyMsg = new Any();
+    anyMsg.setTypeUrl(typeUrl);
+    anyMsg.setValue(bytes);
+    return anyMsg;
+  }
+}
+
+export const wellKnownParsers = new Map([
```

Comment:
Please make sure the list of supported types is the same we have in `base`. Maybe placing a comment with a reference would help both here and there.

Reply:
As discussed, the parsers have no relation to `base` since they were removed from there.